core/linux-dist/x11.cpp

```diff
@@ -117,24 +136,35 @@ void input_x11_handle()
 
 void input_x11_init()
 {
-	x11_keymap[113] = DC_DPAD_LEFT;
-	x11_keymap[114] = DC_DPAD_RIGHT;
+	x11_keymap[KEY_LEFT] = DC_DPAD_LEFT;
+	x11_keymap[KEY_RIGHT] = DC_DPAD_RIGHT;
+
+	x11_keymap[KEY_UP] = DC_DPAD_UP;
+	x11_keymap[KEY_DOWN] = DC_DPAD_DOWN;
 
-	x11_keymap[111] = DC_DPAD_UP;
-	x11_keymap[116] = DC_DPAD_DOWN;
+	// Layout on a real DC controller
+	//   Y
+	// X   B
+	//   A
+	x11_keymap[KEY_S] = DC_BTN_X;
+	x11_keymap[KEY_X] = DC_BTN_A;
+	x11_keymap[KEY_D] = DC_BTN_Y;
+	x11_keymap[KEY_C] = DC_BTN_B;
 
-	x11_keymap[53] = DC_BTN_X;
-	x11_keymap[54] = DC_BTN_B;
-	x11_keymap[55] = DC_BTN_A;
+	// ???
+	x11_keymap[KEY_Q] = DC_BTN_Z;
+	x11_keymap[KEY_W] = DC_BTN_C;
+	x11_keymap[KEY_E] = DC_BTN_D;
+
+	// Start button (triangle)
+	x11_keymap[KEY_RETURN] = DC_BTN_START;
 
 	/*
 	//TODO: Fix sliders
-	x11_keymap[38] = DPad_Down;
-	x11_keymap[39] = DPad_Down;
+	x11_keymap[KEY_A] = DPad_Down;
```

Comment:
I'd introduce some special DC_BTN_* values here and use them. Maybe DC_VIRT_LT to make it clear they do not exist in the h/w?

Reply:
Are the sliders meant to be the shoulder buttons LT and RT?
Then why not just use ``DC_AXIS_LT`` and ``DC_AXIS_LT`` which are defined in https://github.com/reicast/reicast-emulator/blob/a4cf716f99ad8f76b85c24063b12958bdd82f22d/core/linux-dist/main.h#L31